veles/secrets/gcpoauth2token/validator.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcpoauth2token
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/google/osv-scalibr/veles"
+)
+
+// tokenInfoResponse represents the response from Google's tokeninfo endpoint.
+type tokenInfoResponse struct {
+	Audience         string `json:"audience,omitempty"`
+	Scope            string `json:"scope,omitempty"`
+	UserID           string `json:"user_id,omitempty"`
+	ExpiresIn        string `json:"expires_in,omitempty"`
+	Email            string `json:"email,omitempty"`
+	Error            string `json:"error,omitempty"`
```

Comment:
should the response contain this field? Most of documents I found doesn't?
https://cloud.google.com/docs/authentication/token-types#user-access-tokens
https://developers.google.com/resources/api-libraries/documentation/oauth2/v2/java/latest/com/google/api/services/oauth2/model/Tokeninfo.html